packages/framework/src/Component/Image/ImageRepository.php

```diff
@@ -141,4 +145,54 @@ public function getMainImagesByEntitiesIndexedByEntityId(array $entitiesOrEntity
 
         return $imagesByEntityId;
     }
+
+    /**
+     * @param int $imageId
+     * @return \Shopsys\FrameworkBundle\Component\Image\Image|null
+     */
+    public function findById(int $imageId): ?Image
+    {
+        return $this->getImageRepository()->find($imageId);
+    }
+
+    /**
+     * @param string $entityName
+     * @param int $entityId
+     * @param string|null $type
+     * @return int
+     */
+    public function getImagesCountByEntityIndexedById(string $entityName, int $entityId, ?string $type = null): int
+    {
+        $queryBuilder = $this->em->createQueryBuilder()
+            ->select('COUNT(i)')
+            ->from(Image::class, 'i', 'i.id')
+            ->andWhere('i.entityName = :entityName')->setParameter('entityName', $entityName)
+            ->andWhere('i.entityId = :entityId')->setParameter('entityId', $entityId);
+
+        if ($type === null) {
+            $queryBuilder->andWhere('i.type IS NULL');
+        } else {
+            $queryBuilder->andWhere('i.type = :type')->setParameter('type', $type);
+        }
+
+        return $queryBuilder->getQuery()->getSingleScalarResult();
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Image\Image[] $images
+     * @return \Shopsys\FrameworkBundle\Component\Image\Image[]
+     */
+    protected function initTranslations(array $images): array
+    {
+        foreach ($images as &$image) {
+            /** @var \Doctrine\ORM\PersistentCollection $translations */
+            $translations = $image->getTranslations();
+
+            if ($translations instanceof PersistentCollection) {
+                $translations->initialize();
+            }
+        }
+
+        return $images;
+    }
```

Comment:
what is the benefit of this initialization? 

It seems like it always loads a complete collection of translations with an extra query, which is not what we normally do. 
Shouldn't it be eagerly loaded while loading the data from DB?

Reply:
Honestly, I don't know. I just moved functionality from project base to package. I haven't got much information from author of this change on project, because it was quite a long ago :man_shrugging:

Reply:
Do we want to investigate it further ?

Reply:
Moving image functionality from project-base to framework, has been stopped for now. We will resolve it in other issue. The reason is, that the whole image entity is cached and it causes problems with translations, that might be missing. We don't want to think about missing translations. 

Solution in project-base could be called work-around, and it does not belong to framework package. We need to try another approach